src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -0,0 +1,584 @@
+            if (DataManager.useS3) {
+                String s3Key = String.join("/", "project", filename);
+                FeedStore.s3Client.putObject(DataManager.feedBucket, s3Key, mergedTempFile);
+                LOG.info("Storing merged project feed at s3://{}/{}", DataManager.feedBucket, s3Key);
+            } else {
+                try {
+                    FeedVersion.feedStore.newFeed(filename, new FileInputStream(mergedTempFile), null);
+                } catch (IOException e) {
+                    e.printStackTrace();
+                    LOG.error("Could not store feed for project {}", filename);
+                    throw e;
+                }
+            }
+        } else {
+            // Store the zip file for the merged feed version.
+            try {
+                FeedVersion.feedStore.newFeed(mergedVersion.id, new FileInputStream(mergedTempFile), feedSource);
+            } catch (IOException e) {
+                LOG.error("Could not store merged feed for new version");
+                throw e;
+            }
+        }
+    }
+
+    /**
+     * Merge the specified table for multiple GTFS feeds.
+     * @param table table to merge
+     * @param feedsToMerge map of feedSources to zipFiles from which to extract the .txt tables
+     * @return single merged table for feeds or null if the table did not exist for any feed
+     */
+    private byte[] constructMergedTable(Table table, List<FeedToMerge> feedsToMerge) throws IOException {
+        String keyField = table.getKeyFieldName();
+        String orderField = table.getOrderFieldName();
+        if (mergeType.equals(MTC)) {
+            // MTC requires that the stop and route records be merged based on different key fields.
+            switch (table.name) {
+                case "stops":
+                    keyField = "stop_code";
+                    break;
+                case "routes":
+                    keyField = "route_short_name";
+                    break;
+                default:
+                    // Otherwise, use the standard key field (see keyField declaration.
+                    break;
+            }
+        }
+        // Set up objects for outputting the data and tracking the rows encountered
+        ByteArrayOutputStream tableOut = new ByteArrayOutputStream();
+        Map<String, String[]> rowValuesForKeys = new HashMap<>();
+        Set<String> rowStrings = new HashSet<>();
+        int mergedLineNumber = 0;
+        // Get the spec fields to export
+        List<Field> specFields = table.specFields();
+        boolean stopCodeMissingFromFirstTable = false;
+        try {
+            // Iterate over each zip file.
+            for (int feedIndex = 0; feedIndex < feedsToMerge.size(); feedIndex++) {
+                mergeFeedsResult.feedCount++;
+                // FIXME add check for merge type.
+                if (feedIndex > 0 && (table.name.equals("agency") || table.name.equals("feed_info")) && mergeType.equals(MTC)) {
```

Comment:
Please format this line so it isn't so long. Also, please elaborate on why the merge type should be checked in the comment above.